netket/models/rbm.py

```diff
@@ -129,10 +143,65 @@ def __call__(self, x):
         return re + 1j * im
 
 
+class RBMMultiVal(nn.Module):
+    """
+    A fully connected Restricted Boltzmann Machine (see :ref:`netket.models.RBM`) suitable for large local hilbert spaces.
+    Local quantum numbers are passed through a one hot encoding that maps them onto
+    an enlarged space of +/- 1 spins. In turn, these quantum numbers are used with a
+    standard RbmSpin wave function.
+    """
+
+    n_classes: int
+    """The number of classes in the one-hot encoding"""
+    dtype: Any = np.float64
+    """The dtype of the weights."""
+    activation: Any = nknn.logcosh
+    """The nonlinear activation function."""
+    alpha: Union[float, int] = 1
+    """feature density. Number of features equal to alpha * input.shape[-1]"""
+    use_hidden_bias: bool = True
+    """if True uses a bias in the dense layer (hidden layer bias)."""
+    use_visible_bias: bool = True
+    """if True adds a bias to the input not passed through the nonlinear layer."""
+    precision: Any = None
+    """numerical precision of the computation see `jax.lax.Precision`for details."""
+
+    kernel_init: Callable[[PRNGKey, Shape, Dtype], Array] = default_kernel_init
+    """Initializer for the Dense layer matrix."""
+    hidden_bias_init: Callable[[PRNGKey, Shape, Dtype], Array] = default_kernel_init
+    """Initializer for the hidden bias."""
+    visible_bias_init: Callable[[PRNGKey, Shape, Dtype], Array] = default_kernel_init
+    """Initializer for the visible bias."""
+
+    def setup(self):
+        self.rbm = RBM(
+            name="RBM",
```

Comment:
There seems to be a problem with passing `name=` here:
```
Traceback (most recent call last):
  File "Examples/Ising1d/ising1d.py", line 38, in <module>
    gs = nk.VMC(ha, op, sa, ma, n_samples=1000)
  File "<...>/netket/netket/driver/vmc.py", line 56, in __init__
    variational_state = MCState(*args, **kwargs)
  File "<...>/netket/netket/variational/mc_state.py", line 197, in __init__
    self.init(seed, dtype=sampler.dtype)
  File "<...>/netket/netket/variational/mc_state.py", line 226, in init
    variables = self._init_fun({"params": key}, dummy_input)
  File "<...>/netket/netket/variational/mc_state.py", line 165, in <lambda>
    lambda model, *args, **kwargs: model.init(*args, **kwargs), model
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 719, in init
    _, v_out = self.init_with_output(rngs, *args, method=method, **kwargs)
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 704, in init_with_output
    return self.apply(
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 686, in apply
    return apply(fn, mutable=mutable)(variables, rngs=rngs)
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/core/scope.py", line 591, in wrapper
    y = fn(root, *args, **kwargs)
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 685, in <lambda>
    fn = lambda scope: method(self.clone(parent=scope), *args, **kwargs)
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 562, in clone
    return self.__class__(**attrs)
  File "<string>", line 15, in __init__
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 502, in __post_init__
    self.setup()
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 221, in wrapped_module_method
    return fun(self, *args, **kwargs)
  File "<...>/netket/netket/models/rbm.py", line 177, in setup
    self.rbm = RBM(
  File "<...>/netket/.direnv/python-3.8.6/lib/python3.8/site-packages/flax/linen/module.py", line 430, in __setattr__
    raise ValueError(
ValueError: In setup, assign names of Modules via self.<name> and not using keyword argument name="<name>"
```

Reply:
yeah, should have added a bunch of tests.
Did that now.